src/main/scala/com/danielasfregola/twitter4s/TwitterRestClient.scala

```diff
@@ -62,23 +63,50 @@ trait RestClients
     with TwitterUserClient
 
 object TwitterRestClient {
-
   def apply(): TwitterRestClient = {
-    val consumerToken = ConsumerToken(key = consumerTokenKey, secret = consumerTokenSecret)
-    val accessToken = AccessToken(key = accessTokenKey, secret = accessTokenSecret)
-    apply(consumerToken, accessToken)
+    // See what we have to work with
+    authMode match {
+      case UseOAuth1 => {
```

Comment:
We really do not need these` Configurations`, we can infer the configuration from the user setup